src/architecture/utilities/orbitalMotion.c

```diff
@@ -322,27 +267,20 @@ double N2H(double N, double e)
     double small = 1e-13;
     double dH = 10 * small;
     double H1 = N;
-    int    max = 200;
-    int    count = 0;
+    int    maxIterations = 200;
+    int    iterationCount = 0;
     if(fabs(H1) > 7.0)
     {
         H1 = N/fabs(N)*7.0;
     }
 
-    if(e > 1) {
-        while(fabs(dH) > small) {
-            dH = (e * sinh(H1) - H1 - N) / (e * cosh(H1) - 1);
-            H1 -= dH;
-            if(++count > max) {
-                BSK_PRINT(MSG_ERROR, "iteration error in N2H(%f,%f)", N, e);
-                dH = 0.;
-            }
-        }
-    } else {
-        H1 = NAN;
-        BSK_PRINT(MSG_ERROR, "N2H() received e = %g. The value of e should be e > 1.", e);
+    assert(1 < e);
+    while(fabs(dH) > small) {
+        assert(iterationCount <= maxIterations);
```

Comment:
Same concern as my sibling comment: since failing this condition previously just returned the current best approximation of H1, it doesn't seem appropriate to `assert` here. I'd recommend similar code to what I posted earlier in the file.

Reply:
Added 😄